packages/js-sdk/src/datastore/sync.ts

```diff
@@ -155,15 +155,19 @@ export class Sync {
       try {
         multiInsertResult = await network.create(entitiesForInsert, options);
       } catch (error) {
-        // In case of a general error with the batch insert, simulate per-entity errors
-        return syncDocsForInsert.forEach((doc, index) => {
-          totalPushResults.push({
-            _id: doc.entityId,
-            operation: SyncEvent.Create,
-            entity: entitiesForInsert[index],
-            error
+        // In case of a general batch insert error, do not break the push operation and aggregate all errors in the result
+        if (options.catchGenericErrors === true) {
+          return syncDocsForInsert.forEach((doc, index) => {
+            totalPushResults.push({
+              _id: doc.entityId,
```

Comment:
There was some discussion about adding an `index` property - not sure what was decided on that.

Reply:
This should not actually happen anymore because `create` would get the general error and not the array of individual errors.

Reply:
I thought it was an issue which concerned push() as well. Alright then.